Client/Frontend/TabBar/TabBarViewController.swift

```diff
@@ -21,10 +21,10 @@ private let BackgroundColor = UIColor(red: 57.0 / 255, green: 57.0 / 255, blue:
 private let TransitionDuration = 0.25
 
 protocol TabBarViewControllerDelegate {
-    func didEnterURL(url: NSURL)
+    func didEnterURL(url: NSURL?)
```

Comment:
If `didEnterURL` is called with an optional, then what does it mean if url is `nil` ? I think what you originally had is more correct.